### PR TITLE
Use java nio instead of io

### PR DESCRIPTION
**Pull request by alexsuter**
It's now possible to inject Path instead of File in a Mojo. I use now everywhere Path.
Next step, I will try to get rid of commons-io dependency.

### CODE REVIEW

File: src/main/java/ch/ivyteam/ivy/maven/engine/EngineClassLoaderFactory.java [hunk 1]
```diff
@@ -48,94 +49,99 @@
...
+          OsgiDir.INSTALL_AREA.resolve(OsgiDir.LIB_BOOT),
           OsgiDir.PLUGINS,
-          OsgiDir.INSTALL_AREA + "/configuration/org.eclipse.osgi", // unpacked
-                                                                    // jars from
-                                                                    // OSGI
-                                                                    // bundles
-          "webapps" + File.separator + "ivy" + File.separator + "WEB-INF" + File.separator + "lib");
+          // unpacked jars from OSGI bundles
+          OsgiDir.INSTALL_AREA.resolve("configuration").resolve("org.eclipse.osgi"),
+          Path.of("webapps").resolve("ivy").resolve("WEB-INF").resolve("lib"));
```
Comment (ivy-rew) [inline src/main/java/ch/ivyteam/ivy/maven/engine/EngineClassLoaderFactory.java:67]: :sunglasses:

File: src/main/java/ch/ivyteam/ivy/maven/engine/EngineVmOptions.java [hunk 2]
```diff
@@ -16,14 +16,6 @@
 
 package ch.ivyteam.ivy.maven.engine;
 
-public class EngineVmOptions {
-  public final String maxmem;
```
Comment (ivy-rew) [inline src/main/java/ch/ivyteam/ivy/maven/engine/EngineVmOptions.java:20]: no consumers? :thinking:

Comment (alexsuter) [reply src/main/java/ch/ivyteam/ivy/maven/engine/EngineVmOptions.java:20]: wondered too. nobody is using it.

File: src/main/java/ch/ivyteam/ivy/maven/engine/deploy/DeploymentOptionsFileFactory.java [hunk 3]
```diff
@@ -45,20 +46,19 @@ private static boolean isOptionsFile(File optionsFile) {
             optionsFile.canRead();
   }
 
-  public File createFromConfiguration(String options) throws MojoExecutionException {
-    File yamlOptionsFile = getTargetFile(YAML);
+  public Path createFromConfiguration(String options) throws MojoExecutionException {
+    var yamlOptionsFile = getTargetFile(YAML);
     try {
-      FileUtils.write(yamlOptionsFile, options, StandardCharsets.UTF_8);
```
Comment (ivy-rew) [inline src/main/java/ch/ivyteam/ivy/maven/engine/deploy/DeploymentOptionsFileFactory.java:51]: :heavy_check_mark:

File: src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java [hunk 4]
```diff
@@ -194,23 +193,23 @@ public void startEngine_copiedEngine_executable() throws Exception {
...
     }
   }
 
-  private void assertFileExecutable(File file) {
-    assertThat(file).exists();
-    assertThat(file.canExecute()).isTrue();
+  private void assertFileExecutable(Path file) {
+    assertThat(file)
+            .exists()
+            .isExecutable();
```
Comment (ivy-rew) [inline src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java:212]: :checkered_flag:

**Review by ivy-rew — APPROVED** [review]
looks great, did some investigation .... bout to big a change for detailed analysis -> gonna trust your work :)

**Pull request merged**